src/presentation/http/router/router.ts

```diff
@@ -0,0 +1,62 @@
+import { FastifyInstance, FastifyReply, FastifyRequest } from 'fastify';
+import NoteRouter from '@presentation/http/router/noteRouter/noteRouter.js';
+import Router, { Route } from '@presentation/http/router/index.js';
+
+/**
+ * Class representing router
+ */
+export default class RootRouter implements Router {
+  /**
+   * Router prefix
+   */
+  public prefix: string;
+
+  /**
+   * Routes (Action, path, handler)
+   */
+  public routes: Route[] = [];
+
+  /**
+   * Note router
+   */
+  public noteRouter: NoteRouter;
+
+  /**
+   * Creates router instance
+   *
+   * @param prefix - router prefix
+   */
+  constructor(prefix: string) {
+    this.prefix = prefix;
+    this.routes = [
+      ['get', '/', this.index],
+    ];
+    this.noteRouter = new NoteRouter('/note');
+  }
+
+  /**
+   * Root route
+   *
+   * @param request - request object
+   * @param reply - reply object
+   */
+  public async index(request: FastifyRequest, reply: FastifyReply): Promise<void> {
+    reply.send({ hello: 'world' });
+  }
+
+  /**
+   * Registers routes
+   *
+   * @param server - fastify server instance
+   */
+  public register(server: FastifyInstance): void {
```

Comment:
this method could be moved to the abstract class along with `prefix` and `routes` props